src/test/javascript/spec/component/modeling-editor/modeling-editor.component.spec.ts

```diff
@@ -0,0 +1,175 @@
+                { provide: TranslateService, useClass: MockTranslateService },
+                { provide: ActivatedRoute, useValue: route },
+                { provide: Router, useValue: MockRouter },
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+            ],
+            schemas: [],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(ModelingEditorComponent);
+            });
+    });
+
+    afterEach(function () {
+        sandbox.restore();
+    });
+
+    it('ngAfterViewInit', () => {
+        // @ts-ignore
+        fixture.componentInstance.umlModel = testClassDiagram as UMLModel;
+        fixture.detectChanges();
+
+        // test
+        fixture.componentInstance.ngAfterViewInit();
+        expect(fixture.componentInstance['apollonEditor']).toBeTruthy();
+    });
+
+    it('ngOnDestroy', () => {
+        // @ts-ignore
+        fixture.componentInstance.umlModel = testClassDiagram as UMLModel;
+        fixture.detectChanges();
+        fixture.componentInstance.ngAfterViewInit();
+        expect(fixture.componentInstance['apollonEditor']).toBeTruthy();
+
+        // test
+        fixture.componentInstance.ngOnDestroy();
+        expect(fixture.componentInstance['apollonEditor']).toBeFalsy();
+    });
+
+    it('ngOnChanges', () => {
+        // @ts-ignore
+        const model = testClassDiagram as UMLModel;
+        fixture.componentInstance.umlModel = model;
+        fixture.detectChanges();
+        fixture.componentInstance.ngAfterViewInit();
+        expect(fixture.componentInstance['apollonEditor']).toBeTruthy();
+
+        const changedModel = cloneDeep(model);
+        changedModel.elements = [];
+        changedModel.relationships = [];
+        changedModel.interactive = { elements: [], relationships: [] };
+        changedModel.size = { height: 0, width: 0 };
+
+        // test
+        fixture.componentInstance.ngOnChanges({
+            umlModel: {
+                currentValue: changedModel,
+                previousValue: model,
+            } as SimpleChange,
+        });
+
+        expect(fixture.componentInstance['apollonEditor']!.model).toEqual(changedModel);
+    });
+
+    it('isFullScreen false', () => {
+        // test
+        const fullScreen = fixture.componentInstance.isFullScreen;
+        expect(fullScreen).toBeFalsy();
+    });
+
+    it('getCurrentModel', () => {
+        // @ts-ignore
+        fixture.componentInstance.umlModel = testClassDiagram as UMLModel;
+        fixture.detectChanges();
+        fixture.componentInstance.ngAfterViewInit();
+        expect(fixture.componentInstance['apollonEditor']).toBeTruthy();
+
+        // test
+        const model = fixture.componentInstance.getCurrentModel();
+        // TODO: uncomment after deserialization bugfix in Apollon libary
```

Comment:
which deserialization issue do you mean? can you reference it here?
Is it easy to fix or more complex?

Reply:
https://github.com/ls1intum/Apollon/issues/146 
Not sure about the complexity, I invested a bit of time and located the issue, but did not yet fully understand why it happens. Can post these information in the Apollon ticket